src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION = 0;
+    private static final int AMOUNT = 1;
+    private static final String REGEX = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private StringBuilder stringBuilder;
+    private File file;
+    private int sumSupply = 0;
+    private int sumBuy = 0;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        readTheFile(fromFileName);
+        countSum();
+        writeTheFile(toFileName);
+    }
+
+    private void readTheFile(String fromFileName) {
+        file = new File(fromFileName);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("can`t read the file" + fromFileName, e);
+        }
+    }
+
+    private void writeTheFile(String toFileName) {
+        file = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file, true))) {
+            stringBuilder = new StringBuilder().append(SUPPLY).append(REGEX).append(sumSupply)
```

Comment:
This method should only write to the file, this logic should be in the method where you generate the result